File: ruby/easy/number_of_segments_in_a_string.rb
# @param {String} s
# @return {Integer}
def count_segments(s)
    i, str = 0, ""
    arr = Array.new
    while i < s.length
        if s[i] != ' '
            str += s[i]
        else
            arr.push(str) unless str.empty?
            str = ""
        end
        i += 1
    end
    arr.push(str) unless str.empty?
    return arr.length
end

File: javascript/easy/numberofSegmentsinaString.js
/**
 * @param {string} s
 * @return {number}
 */
var countSegments = function(s) {
    let count = 0, numSegments = 0;
    for(let c of s) {
        if(c !== ' ') {
            count = 1;
        } else {
            numSegments += count;
            count = 0;
        }
    }
    return numSegments + count;
};
